sycl/source/detail/bindless_images.cpp

```diff
@@ -28,11 +28,25 @@ void populate_pi_structs(const image_descriptor &desc, pi_image_desc &piDesc,
   piDesc.image_width = desc.width;
   piDesc.image_height = desc.height;
   piDesc.image_depth = desc.depth;
-  piDesc.image_type = desc.depth > 0 ? PI_MEM_TYPE_IMAGE3D
-                                     : (desc.height > 0 ? PI_MEM_TYPE_IMAGE2D
-                                                        : PI_MEM_TYPE_IMAGE1D);
+
+  if (desc.array_size > 1) {
+    // Image Array
+    if (desc.depth > 0) {
+      // Image arrays must be 1D or 2D
+      throw sycl::exception(sycl::make_error_code(sycl::errc::invalid),
```

Comment:
Is it missing support or they simply can't be per the specification? If the latter, why can't we rely on `verify` to throw instead? Same in `handler.cpp`.